cmd/agent/entrypoint.go

```diff
@@ -273,6 +286,23 @@ func (ep *Entrypoint) Start() error {
 		ep.srv.Close()
 	})
 
+	{
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		g.Add(func() error {
+			// Wait just a litle bit of time to start the cluster so that our APIs
+			// and other components are running, allowing peers to immediately invoke
+			// them.
+			time.Sleep(100 * time.Millisecond)
+			ep.node.Start()
```

Comment:
This seems like a bit of a code smell, and is obscuring a bigger underlying problem. The original intent was to protect peers from invoking HTTP or gRPC endpoints before the listeners are open. (Note that that wouldn't actually happen, since the ports are open prior to this goroutine ever running, but that's beside the point). 

However, there generally may be transient HTTP or gRPC disconnects, and there is going to need to be a recommended way of handling that. 

If a function needs to deliver a message to its owner, but the owner is not responding, the function has two choices:

- Return a failure (either immediately or after a retry period)
- Failover to the next owner 

Failover requires specific scenarios to be acceptable; if you are transmitting stateful data, you would not want to do premature failover before an unhealthy node is completely evicted from the cluster. Generally, returning a failure is the safest approach, where the caller would either retry or drop the failed request. 

rfratto/ckit may want to extend itself to support node health checks to further filter down the memberlist to the set of working nodes.